packages/protocol/test/governance/double_signing_slasher.ts

```diff
@@ -0,0 +1,160 @@
+    )
+  })
+
+  describe('#initialize()', () => {
+    it('should have set the owner', async () => {
+      const owner: string = await slasher.owner()
+      assert.equal(owner, accounts[0])
+    })
+    it('should have set slashing incentives', async () => {
+      const res = await slasher.slashingIncentives()
+      assert.equal(res[0].toNumber(), 10000)
+      assert.equal(res[1].toNumber(), 100)
+    })
+    it('can only be called once', async () => {
+      await assertRevert(slasher.initialize(registry.address, 10000, 100))
+    })
+  })
+
+  describe('#setSlashingIncentives()', () => {
+    it('can only be set by the owner', async () => {
+      await assertRevert(slasher.setSlashingIncentives(123, 67, { from: nonOwner }))
+    })
+    it('reward cannot be larger than penalty', async () => {
+      await assertRevert(slasher.setSlashingIncentives(123, 678))
+    })
+    it('should have set slashing incentives', async () => {
+      await slasher.setSlashingIncentives(123, 67)
+      const res = await slasher.slashingIncentives()
+      assert.equal(res[0].toNumber(), 123)
+      assert.equal(res[1].toNumber(), 67)
+    })
+    it('should emit the corresponding event', async () => {
+      const resp = await slasher.setSlashingIncentives(123, 67)
+      assert.equal(resp.logs.length, 1)
+      const log = resp.logs[0]
+      assertContainSubset(log, {
+        event: 'SlashingIncentivesSet',
+        args: {
+          penalty: new BigNumber(123),
+          reward: new BigNumber(67),
+        },
+      })
+    })
+  })
+
+  describe('#slash()', () => {
+    const blockNumber = 100
+    const validatorIndex = 5
+    const blockA = ['0x12', '0x12', '0x12']
+    const blockB = ['0x13', '0x13', '0x13']
+    const blockC = ['0x11', '0x13', '0x14']
+    beforeEach(async () => {
+      await slasher.setBlockNumber(blockA, blockNumber)
+      await slasher.setBlockNumber(blockB, blockNumber + 1)
+      await slasher.setBlockNumber(blockC, blockNumber)
+      await slasher.setEpochSigner(Math.floor(blockNumber / EPOCH), validatorIndex, validator)
+      await slasher.setEpochSigner(Math.floor(blockNumber / EPOCH), validatorIndex + 1, validator)
+      // Signed by validators 0 to 5
+      const bitmap = '0x000000000000000000000000000000000000000000000000000000000000003f'
+      await slasher.setNumberValidators(7)
+      await slasher.setVerifiedSealBitmap(blockA, bitmap)
+      await slasher.setVerifiedSealBitmap(blockB, bitmap)
+      await slasher.setVerifiedSealBitmap(blockC, bitmap)
+    })
+    it('fails if block numbers do not match', async () => {
+      await assertRevert(
+        slasher.slash(validator, validatorIndex, blockA, blockB, 0, [], [], [], [], [], [])
+      )
+    })
+    it('fails if is not signed at index', async () => {
+      await assertRevert(
+        slasher.slash(validator, validatorIndex + 1, blockA, blockC, 0, [], [], [], [], [], [])
```

Comment:
I must be missing something, I would have expected this to pass given the previous line:

```
      await slasher.setEpochSigner(Math.floor(blockNumber / EPOCH), validatorIndex + 1, validator)
```

Reply:
`setEpochSigner` sets the elected signer for epoch.